server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -260,6 +260,24 @@ class ArrivalsGraphStage(name: String = "",
       mergeBestFieldsFromSources(baseArrival, mergeArrival(key).getOrElse(baseArrival))
     }
 
+
+    def scheduleTimeApproximation(searchKey: UniqueArrival, baseLiveArrival: Arrival, logMessage: Boolean = false): Option[UniqueArrival] = {
+      val potentialLiveKey: Option[UniqueArrival] = liveArrivals.keys.collectFirst {
```

Comment:
Were we going to check for multiple matches and log a warning if more than one is found?

Reply:
May be I should sort in that case and get the latest one in a lot .  I just made assumption we would only one always .